lib/kb_orthofinder/kb_orthofinderImpl.py

```diff
@@ -277,31 +278,41 @@ def annotate_plant_transcripts(self, ctx, input):
         if(len(sequences_dict)==0):
             raise Exception("The genome does not contain any protein sequences!")
 
+        # We have a problem with how OrthoFinder arbitrarily replaces 'special' characters in identifiers:
+        # https://github.com/davidemms/OrthoFinder/blob/master/scripts_of/util.py#L181
+        # without 'hacking' the OF code, here I'm trying to anticipate how OF may have changed the identifier
+
+        alt_seq_ids_dict=dict()
+        for og_id in sequences_dict:
+            of_id = og_id.replace(":", "_").replace(",", "_").replace("(", "_").replace(")", "_")
+            alt_seq_ids_dict[of_id]=og_id
+
         output['ftrs'] = len(sequences_dict.keys())
 
         #Create directory for storing new fasta file
-        uuid_string = str(uuid.uuid4())
+        uuid_string = "print_fasta_"+str(uuid.uuid4())
```

Comment:
would it be helpful to have all directories from a single run have the same UUID?

Reply:
Yes, that's a better idea but I'll implement it the next time round

Reply:
If you are anything like me, things to be implemented "next time round" end up never being done... I have left many `TODO` comments with sensible code improvements that I aspired to implement but never did! :)

Reply:
It shouldn't be too much work to do this now -- just generate the UUID once and pass it on to all the places that you currently generate a new UUID.

Reply:
Updated in 9ca841d60089414785b076375ae088febb9d0668